### ISSUE TITLE
Copyright assignment to the FluentMigrator project

### ISSUE THREAD

**Issue by fubar-coder** [fluentmigrator/fluentmigrator#841]
Hi,

I plan to change the copyright notice of all source code files to:

```
// Copyright (c) 2018, Fluent Migrator Project
```

The intention behind this change is to avoid different copyright claims on different parts of the project, which should avoid legal problems in the future. You can read more about copyright assignment and the FSF in its [Why the FSF gets copyright assignments from contributors](http://www.gnu.org/licenses/why-assign.html) document.

I already contacted Sean Chambers and asked for his permission.

@garchibald There are 4 source code files containing your copyright. Are you willing to assign the copyright to the Fluent Migrator Project too?

EDIT: Some links for CLA/CAA generation:
- http://contributoragreements.org/ca-cla-chooser/
- https://www.clahub.com/
- http://selector.harmonyagreements.org/

**Comment by orck-adrouin** [comment]
Will you setup a CLA bot to automatically assign copyright to the project?  I use https://www.clahub.com in one of my project and it works quite well.

**Comment by fubar-coder** [comment]
That sounds like a good idea.

**Comment by fubar-coder** [comment]
This [article from Ben Balter](https://ben.balter.com/2018/01/02/why-you-probably-shouldnt-add-a-cla-to-your-open-source-project/) suggests, that a CLA (or CAA) might not be needed - and even hostile for contributors.

**Comment by orck-adrouin** [comment]
Interesting article.  It depends what the goal of the project is in regard to copyright.  If you want to have "copyright 20XX, Fluent Migrator Project" on every files, your contributors need to transfer their ownership to you otherwise the copyright is theirs.  Another thing to consider is this: should the copyright notice of each files list every contributor and their copyrights?

Something like this is a bit excessive but I'm sure you get the point:
```
(c) 2018 Fluent Migrator Project
(c) 2017 John Smith (because he changed method X to do Y)
(c) 2016 Jane Doe (because she added functionality Z)
```

As you can see this can become a bit ridiculous after a few years depending on the number of contributors.  In my experience with open source project CLA are not really a problem but I agree that they can seem intimidating for young developers.  You can simplify the wording of the agreement if necessary to make it more friendly.

**Comment by schambers** [comment]
@jzabroski I'd like to pick this up in the near future. mostly housekeeping but I think we should remove my name from all code files and replace with "FluentMigrator Project". Thoughts?

**Comment by jzabroski** [comment]
Yes, please

**Comment by schambers** [comment]
For your review @jzabroski: https://github.com/fluentmigrator/fluentmigrator/pull/1745
